addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/handler/SqueezeBoxNotificationPlayer.java

```diff
@@ -0,0 +1,264 @@
+    private String mac;
+
+    boolean playlistModified;
+
+    private int notificationMessagePlaylistsIndex;
+
+    SqueezeBoxNotificationPlayer(SqueezeBoxPlayerHandler squeezeBoxPlayerHandler,
+            SqueezeBoxServerHandler squeezeBoxServerHandler, StringType uri) {
+        this.squeezeBoxPlayerHandler = squeezeBoxPlayerHandler;
+        this.squeezeBoxServerHandler = squeezeBoxServerHandler;
+        this.mac = squeezeBoxPlayerHandler.getMac();
+        this.uri = uri;
+        this.playerState = new SqueezeBoxPlayerState(squeezeBoxPlayerHandler);
+    }
+
+    void play() throws InterruptedException, SqueezeBoxTimeoutException {
+        if (squeezeBoxServerHandler == null) {
+            logger.warn("Server handler is null");
+            return;
+        }
+
+        setupPlayerForNotification();
+
+        try {
+            addNotificationMessageToPlaylist();
+            playNotification();
+        } finally {
+            if (playlistModified) {
+                // Mute the player to prevent any noise during the transition
+                // to previous state.
+                setVolume(0);
+                removeNotificationMessageFromPlaylist();
+            }
+        }
+    }
+
+    @Override
+    public void close() {
+        restorePlayerState();
+    }
+
+    private void setupPlayerForNotification() throws InterruptedException, SqueezeBoxTimeoutException {
+        logger.debug("Setting up player for notification");
+        if (!playerState.isPoweredOn()) {
+            logger.debug("Powering on the player");
+            squeezeBoxServerHandler.powerOn(mac);
+        }
+        if (playerState.isShuffling()) {
+            logger.debug("Turning off shuffle");
+            squeezeBoxServerHandler.setShuffleMode(mac, 0);
+        }
+        if (playerState.isRepeating()) {
+            logger.debug("Turning off repeat");
+            squeezeBoxServerHandler.setRepeatMode(mac, 0);
+        }
+        if (playerState.isPlaying()) {
+            squeezeBoxServerHandler.stop(mac);
+        }
+
+        int notificationVolume = squeezeBoxPlayerHandler.getNotificationSoundVolume().intValue();
+        if (notificationVolume == 0) {
+            logger.warn("Player volume is 0.");
+        }
+        setVolume(notificationVolume);
+    }
+
+    /**
+     * Sends a volume set command if target volume is not equal to the current volume.
+     *
+     * @param requestedVolume The requested volume value.
+     * @throws InterruptedException Thread interrupted during while we were waiting for an answer from the media server.
+     * @throws SqueezeBoxTimeoutException Volume command was not acknowledged by the media server.
+     */
+    private void setVolume(int requestedVolume) throws InterruptedException, SqueezeBoxTimeoutException {
+        if (playerState.getVolume() == requestedVolume) {
+            return;
+        }
+
+        SqueezeBoxNotificationListener listener = new SqueezeBoxNotificationListener(mac);
```

Comment:
It feels heavy to construct a listener just for this goal and detach it later it seems you even do it twice in methods called from the non-private method (being `play`).

Correction at least 4 times...

Reply:
Yes - indeed; I just took that from the original code and would not like to touch that mechanism at the moment.

Reply:
It is a little heavy -- not sure who did that.  :-o  

But I agree that we'd best not change it ATM.